extensions/amp-form/0.1/amp-form.js

```diff
@@ -626,7 +626,10 @@ export class AmpForm {
    * @private
    */
   doActionXhr_() {
-    return this.doXhr_(dev().assertString(this.xhrAction_), this.method_);
+    // Since action-xhr can be modified by linker, don't use cached version.
+    return this.doXhr_(
+        dev().assertString(this.form_.getAttribute('action-xhr')),
```

Comment:
Should we be using `getXhrUrl_('action-xhr')` here or do we not require the assert HTTPS and assert not on AMP CDN checks at this point? Random question: Why can't the linker also update `this.xhrAction_`?

Reply:
My thought was that at this point the original `action-xhr` has already been verified by `getXhrUrl_`, but I think you are right it is safer to also call it on the new linker version.

Right now the form-submit service is only sending the form element to the callback. I guess it could be refactored to send the impl to change `this.xhrAction_`. I think it is simpler to only check and manipulate the actual DOM element, but open to changing if you feel strongly.

Reply:
Changed to `getXhrUrl_`.

Reply:
Another way to do this would be to have a public setter that allows setting `this.xhrAction_`--this way we would have one consistent representation of `action-xhr`, and not have two different values. I guess the main question here is after you update `action-xhr` with the Linker, do we need to keep the previous value of `this.xhrAction_` around (is it used for anything instead of the updated attribute)? If yes, we do need two different values, then maybe add a comment explaining why. If not, then maybe we should only keep one consistent value.

Reply:
Talked with @cathyxz offline. Discussed a few options. 1. Adding setter - this would require passing the impl through the form-submit service. 2. Overwrite `action-xhr` - this could lead to a weird state where `AmpForm.xhrAction_` is different than the attribute on the actual form. 3. Add a new linker specific attr, and check to see if that exists on submission. Otherwise, fallback to cached `AmpForm.xhrAction_`

We chose option 3 :)

Reply:
Added new attribute, Done.

Reply:
3. introduced a reverse dependency, which we tried hard to avoid (that's why we had a generic submit service). `amp-form` should not be aware of the existence of linker.

Reply:
Not wanting a reverse dependency makes sense. I am having trouble understanding the difference between 4 and 1?